sympy/geometry/util.py

```diff
@@ -684,23 +681,38 @@ def intersection(*entities):
     """
     from .entity import GeometryEntity
     from .point import Point
-
+    pairwise = kwargs.pop('pairwise', False)
     if len(entities) <= 1:
         return []
-
-    # entities may be an immutable tuple
+    if not pairwise:
+
+        # entities may be an immutable tuple
```

Comment:
This code converting tuples to Points needs to be moved out of the if-block. 

You can verify by using the test I gave before but put `(0,0)` as the first point -- no error should raise.